--- a/scripts/feeds
+++ b/scripts/feeds
@@ -260,8 +260,10 @@ sub list_feed {
 
 sub list {
 	my %opts;
+	my @pkg_list;
+	my %pkg_seen;
 
-	getopts('r:d:sh', \%opts);
+	getopts('r:d:shi', \%opts);
 	if ($opts{h}) {
 		usage();
 		return 0;
@@ -289,6 +291,31 @@ sub list {
 		}
 		return 0;
 	}
+	if ($opts{i}) {
+		get_installed();
+		my $pkg;
+		my $src;
+		foreach my $name (keys %installed) {	
+			$pkg = $installed{$name};
+			$src = $pkg->{src};
+
+			# Skip all seen and installed core packages
+			if (!$pkg_seen{$src} and !is_core_package($src)) {
+				push @pkg_list, ${src};
+				$pkg_seen{${src}} = 1;
+			}
+		}
+
+		my $pkg_version;
+		foreach my $name (sort { lc($a) cmp lc($b) } @pkg_list) {	
+			$pkg = $installed{$name};
+			$pkg_version = $pkg->{version};
+			$pkg_version or $pkg_version = "";
+
+			printf "Installed: $name (%s)\n", ${pkg_version};
+		}
+		return 0;
+	}
 	foreach my $feed (@feeds) {
 		list_feed($feed->[1], @ARGV) if (!defined($opts{r}) or $opts{r} eq $feed->[1]);
 	}
@@ -605,6 +632,7 @@ Commands:
 	list [options]: List feeds, their content and revisions (if installed)
 	Options:
 	    -s :            List of feed names and their URL.
+	    -i :            List non-core installed packages.
 	    -r <feedname>:  List packages of specified feed.
 	    -d <delimiter>: Use specified delimiter to distinguish rows (default: spaces)
 
